dhcp/dhcp.py

```diff
@@ -192,7 +193,16 @@ def received_dhcp_discover(self, discovery):
         if should_send_offer:
             self.send_offer(discovery)
         else:
-            logger.error("unknown mac_address {}. will not assign ip.".format(discovery.client_mac_address))
+            unknown_mac_addr = discovery.client_mac_address
+            logger.error("unknown mac_address {}. will not assign ip.".format(unknown_mac_addr))
+            self.mac_mapper(unknown_mac_addr)
+
+    def mac_mapper(self, unknown_mac):
+        with open('addr_database.csv', 'r') as f:
```

Comment:
you should pull this from the pyz like tftp does. also you should check in this file and document how to update it.

Reply:
Hi Ben, could you elaborate on what you mean by pulling from the pyz like tftp does? My adaptation just replaces what happens when you start looking for DHCP discovers

Reply:
addr_database.csv is a resource, so it should be built into the pyz file and accessed like a resource: https://github.com/SJSU-CS-systems-group/piman/blob/master/tftp/tftp.py#L43

Reply:
added new commit to resolve this issue

Reply:
sorry, where is the commit?

Reply:
https://github.com/SJSU-CS-systems-group/piman/pull/13/commits/4724ba7744dc13781ee28dea2da8bc293820581a